docs/developer-guide/ksqldb-clients/java-client.md

```diff
@@ -113,6 +114,12 @@ Start by creating a `pom.xml` for your Java application:
       by replacing the repositories in the example POM above with a repository with this
       URL instead. Also update `ksqldb.version` to be a CP version, such as `{{ site.cprelease }}`, instead.
 
+!!! note
+      The `with-dependencies` artifact was introduced in ksqlDB version 0.29 and CP version 7.4.0.  This jar includes  
+      all the necessary dependencies and relocates most of them in an attempt to avoid classpath issues.
+      If you want more control over the dependencies on the classpath, you can depend directly on the client 
```

Comment:
How does this work exactly? Does this mean the user would have to ensure that all the Java client's dependencies are available on the classpath via other methods? When would a user prefer to do that rather than using the shaded dependencies, given that those dependencies are already shaded in order to avoid name conflicts?

Reply:
If a user depends on `io.confluent.ksql:ksqldb-api-client` without the `with-dependencies` classifier (e.g.)
```        <dependency>
            <groupId>io.confluent.ksql</groupId>
            <artifactId>ksqldb-api-client</artifactId>
            <version>${ksqldb.version}</version>
        </dependency>```
then yes, they would be responsible for getting all the other dependencies.

A user may prefer to do that if their Java project was using one or more of the bundled dependencies and they wanted to use that dependency as common code.  They may wish to do that to reduce the size of the classpath; that could matter if one were building up a Docker image.

Basically, they'd be taking on the risk of classpath conflicts in order to reduce classpath size.